### PR TITLE
Python package for OpenEI EULP downloads

### PR DESCRIPTION
**Pull request by cpaulgilman**
Add Python package to SAM samples folder to provide a way for SAM users to get load data from OpenEI End User Load Profiles database to use in SAM.

Partially address #539

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] This change requires a documentation update
- [ ] This change modifies variables in existing compute modules. Please see [Checking for PySAM Incompatible API Changes](https://github.com/NREL/SAM/wiki/Checking-for-PySAM-Incompatible-API-Changes).

## Checklist:

If you have added a new compute module in a SSC pull request related to this one, be sure to check the [Process Requirements](https://github.com/NREL/SAM/wiki/Compute-modules-in-SAM).

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [X] I have checked my code and corrected any misspellings

### CODE REVIEW

**Review by brtietz — CHANGES_REQUESTED** [review]
Code looks well organized and well commented, but I got bogged down testing with missing dependencies. Please either add a complete list of dependencies to the README or add a conda environment yml file:

image

image

I'll take another run at testing once that's there.

File: samples/Python Scripts/End Use Load Profiles/readme.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+The `eulp` Python package contains three modules that you can use in your code to work with individual building time series data from the <a href="https://data.openei.org/submissions/4520">OpenEI End Use Load Profiles (EULP) for the U.S. Building Stock</a>:
+
+* `get`: Functions to download metadata, weather, and load files from the various datasets in the database.
+
+* `plot`: Functions to generate plots of the downloaded data.
+
+* `stats`: Functions to calculate total and peak load values from the downloaded data.
+
+To use the package, copy the `eulp` folder to the folder containing your Python script, and import the package with:
```
[the remark this answers is not in the document]
Comment (cpaulgilman) [reply samples/Python Scripts/End Use Load Profiles/readme.md:11]: Good point. How about adding a requirements.txt file along with a list of required packages in the readme? See https://github.com/NREL/SAM/pull/1300/commits/74da475099d613ec51814b39f42d471674bfef32.

[the remark this answers is not in the document]
Comment (brtietz) [reply samples/Python Scripts/End Use Load Profiles/readme.md:11]: Thanks, I added two additional packages I needed to run it.

File: samples/Python Scripts/End Use Load Profiles/load-data-example.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+fname = f'{path}/load-data-all-{bldg_id}.csv'
+
+df = pd.read_csv(fname, 
+                 parse_dates=True, 
+                 index_col='timestamp')
+
+D = plot.clean_plot_data(df, bldg_id) 
+df = D['df_clean']
+
+plot.plot_full_year(df.filter(items=['total']), bldg_id, 'Total Load') # total load over year
```
Comment (brtietz) [inline samples/Python Scripts/End Use Load Profiles/load-data-example.py:118]: I like having plots here, but can we add units. I think its kW, but having the explicit axis labels would be super helpful.

Comment (cpaulgilman) [reply samples/Python Scripts/End Use Load Profiles/load-data-example.py:118]: Yes, thanks for pointing out that omission. I added a "kW" label to the y-axis for all plots.

**Review by brtietz — APPROVED** [review]
Thanks for adding the requirements and labels! Ready to merge.

**Comment by brtietz** [comment]
@cpaulgilman The test failures look like they're due to a problem that's already been solved on patch. I think I might be able to fix it by cherry-picking the commits from 12/9 and 12/12 to a branch from the latest patch. Any objections to closing this PR and trying that approach?

**Comment by brtietz** [comment]
I've moved the commits that aren't merge commits to https://github.com/NREL/SAM/pull/1331. Fingers crossed that the tests pass, closing this with a sense of optimism.

**Pull request closed without merging**